### PR TITLE
dotCMS/core#24368 [UI] Disable Varian Layout if experiment is running. 

### LINKED ISSUE #24368: [UI] Disable Varian Layout if experiment is running.
If the experiment is running, the layout tab should be disabled 

https://user-images.githubusercontent.com/31667212/225066242-362ae3a7-a88e-4ed2-80ef-16563fcf4c04.mov

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Move the running experiment to `DotPageRenderState`
* Disable layout in the `dot-edit-page-nav.component.ts` when the user is on a experiment variant and the experiment is running.

### CODE REVIEW

**Review by zJaaal — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.html [hunk 1]
```diff
@@ -15,7 +15,7 @@
         class="dot-edit__toolbar"
         [pageState]="pageState"
         [variant]="variantData | async"
-        [runningExperiment]="runningExperiment$ | async"
+        [runningExperiment]="pageState.state.runningExperiment"
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.html:18]: Ohhh this is part of the page state now.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts [hunk 2]
```diff
@@ -260,26 +264,20 @@ export class DotPageStateService {
...
-                                }),
-                                switchMap((response: ESContent) => {
-                                    const favoritePage = response.jsonObjectView?.contentlets[0];
-
-                                    return this.setLocalPageState(page, favoritePage);
-                                })
-                            );
+                        return forkJoin([
+                            this.getFavoritePage(user, urlParam),
+                            this.getRunningExperiment(page.page.identifier)
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts:269]: What happen if any of this fails?

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/main/dot-edit-page-nav/dot-edit-page-nav.component.ts [hunk 3]
```diff
@@ -150,4 +151,13 @@ export class DotEditPageNavComponent implements OnChanges {
             !template ? 'editpage.toolbar.nav.layout' : 'editpage.toolbar.nav.layout'
         );
     }
+
+    private disableLayoutOnExperimentVariant(): boolean {
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/main/dot-edit-page-nav/dot-edit-page-nav.component.ts:155]:
```suggestion
    private isDisableLayoutOnExperimentVariant(): boolean {
```

???

File: core-web/libs/dotcms-models/src/lib/dot-rendered-page-state.model.ts [hunk 4]
```diff
@@ -13,13 +15,15 @@ export interface DotPageState {
...
     mode: DotPageMode;
+    runningExperiment: DotExperiment;
 }
 
 export class DotPageRenderState extends DotPageRender {
     constructor(
         private _user: User,
         private dotRenderedPage: DotPageRenderParameters,
-        _favoritePage?: DotCMSContentlet
+        _favoritePage?: DotCMSContentlet,
```
[the remark this answers is not in the document]
Comment (fmontes) [reply core-web/libs/dotcms-models/src/lib/dot-rendered-page-state.model.ts:25]: Maybe... but they're not touching that so let's keep it as it is

**Review by fmontes — DISMISSED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.html (hunk 1 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/dot-edit-content.component.html:18]: yea the other option was to make multiple request, to check if disable the layout menu also to show/hide the running indicator at the top of the edit mode and any other restriction that can happen based on experiments status.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts (hunk 2 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/services/dot-page-state/dot-page-state.service.ts:269]: they will show a inline notification and in this forkjoin will return null.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/main/dot-edit-page-nav/dot-edit-page-nav.component.ts (hunk 3 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/main/dot-edit-page-nav/dot-edit-page-nav.component.ts:155]: in this case I like more `disableLayoutOnExperimentVariant` to keep it as a question based on the conditions inside the function.

**Review by oidacra — APPROVED** [review]

**Pull request merged**